ludwig/schema/hyperopt/scheduler.py

```diff
@@ -0,0 +1,509 @@
+
+    max_t: int = max_t_alias(default=81)
+
+    reduction_factor: int = schema_utils.NonNegativeFloat(
+        default=3, description=("Used to set halving rate and amount. This is simply a unit-less scalar.")
+    )
+
+    stop_last_trials: bool = schema_utils.Boolean(
+        default=True, description=("Whether to terminate the trials after reaching max_t. Defaults to True.")
+    )
+
+
+# TODO: Double-check support for this
+@register_scheduler_config("fifo")
+@dataclass
+class FIFOSchedulerConfig(BaseSchedulerConfig):
+    """FIFO trial scheduler settings."""
+
+    type: str = schema_utils.StringOptions(options=["fifo"], default="fifo", allow_none=False)
+
+
+# TODO: Double-check support for this as well as whether Callable args work properly
+@register_scheduler_config("resource_changing")
+@dataclass
+class ResourceChangingSchedulerConfig(BaseSchedulerConfig):
+    """Resource changing scheduler settings."""
+
+    type: str = schema_utils.StringOptions(options=["resource_changing"], default="resource_changing", allow_none=False)
+
+    base_scheduler: Union[str, None, Callable] = schema_utils.String(
+        description=("The scheduler to provide decisions about trials. If None, a default FIFOScheduler will be used.")
+    )
+
+    resources_allocation_function: Union[str, Callable] = schema_utils.String(
+        description=(
+            "The callable used to change live trial resource requiements during tuning. This callable will be called on"
+            " each trial as it finishes one step of training. The callable must take four arguments: TrialRunner, "
+            "current Trial, current result dict and the ResourceChangingScheduler calling it. The callable must return "
+            "a PlacementGroupFactory, Resources, dict or None (signifying no need for an update). If "
+            "resources_allocation_function is None, no resource requirements will be changed at any time. By default, "
+            "DistributeResources will be used, distributing available CPUs and GPUs over all running trials in a robust"
+            " way, without any prioritization."
+        )
+    )
+
+
+def get_scheduler_conds():
+    """Returns a JSON schema of conditionals to validate against scheduler types defined in
+    `ludwig.schema.hyperopt.scheduler_registry`."""
+    conds = []
+    for scheduler_config in scheduler_config_registry:
+        scheduler_cls = scheduler_config_registry[scheduler_config]
+        other_props = schema_utils.unload_jsonschema_from_marshmallow_class(scheduler_cls)["properties"]
+        schema_utils.remove_duplicate_fields(other_props)
+        preproc_cond = schema_utils.create_cond(
+            {"type": scheduler_config},
+            other_props,
+        )
+        conds.append(preproc_cond)
+    return conds
+
+
+def SchedulerDataclassField(default={"type": "fifo"}, description="Hyperopt scheduler settings."):
```

Comment:
@arnavgarg1 (and CC @ShreyaR ) I have set the default scheduler to `fifo` as requested in a prior comment (https://github.com/ludwig-ai/ludwig/pull/2346#discussion_r1028051666).